### PR TITLE
Steps with strategy pattern

### PR DESCRIPTION
**Pull request by GustavoZHA**

### CODE REVIEW

File: src/main/java/core/selenium/WebElementAction.java [hunk 1]
```diff
@@ -170,4 +170,15 @@ public String getTextOfByFieldByLocator(final By locator) {
...
+
+
+    /**
+     * Gets the text of a web element.
+     *
+     * @param field web element to get text.
+     * @return web element's text
+     */
+    public void clickByXpath(final String field) {
+        WebDriverManager.getInstance().getWebDriver().findElement(By.xpath(field)).click();
```
Comment (JorgeCaceres-Jala) [inline src/main/java/core/selenium/WebElementAction.java:182]: you can use driver.findElement(By.xpath(field)).click(); i

File: src/main/java/salesforce/ui/pages/lightning/worktype/NewWorkTypePage.java [hunk 2]
```diff
@@ -11,11 +11,17 @@
 import core.selenium.WebDriverManager;
 import org.openqa.selenium.By;
 import salesforce.ui.pages.BasePage;
+import salesforce.utlis.strategy.IFeatureNew;
+import salesforce.utlis.supplier.VoidSupplier;
+
```
Comment (JorgeCaceres-Jala) [inline src/main/java/salesforce/ui/pages/lightning/worktype/NewWorkTypePage.java:16]: This line should be deleted, this mistake is repeated in many classes

File: src/test/java/org/salesforce/hooks/generalHooks.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import salesforce.config.EnvConfig;
+import salesforce.ui.PageTransporter;
+
+public class generalHooks {
+    public PageTransporter pageTransporter = new PageTransporter();
+    public SoftAssert softAssert = new SoftAssert();
+
+    @Before(value = "@CreateWorkType")
+    public void setup() {
+        System.out.println("8787878787878787878");
```
Comment (JorgeCaceres-Jala) [inline src/test/java/org/salesforce/hooks/generalHooks.java:16]: This sout should be deleted

File: src/main/java/salesforce/utlis/strategy/IFeaturesPage.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Copyright (c) 2021 Fundacion Jala.
+ * This software is the confidential and proprietary information of Fundacion Jala
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Fundacion Jala
+ */
+
```
Comment (cristianfchq) [inline src/main/java/salesforce/utlis/strategy/IFeaturesPage.java:8]: The name of package should be utils and not utlis

File: src/main/java/salesforce/ui/pages/lightning/contracts/CreatedContractPage.java [hunk 5]
```diff
@@ -9,13 +9,17 @@
 package salesforce.ui.pages.lightning.contracts;
 
 import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
```
Comment (SaulCM) [inline src/main/java/salesforce/ui/pages/lightning/contracts/CreatedContractPage.java:14]: Care with blank lines after formating with Intellij.

File: src/main/java/salesforce/ui/pages/lightning/contracts/NewContractPage.java [hunk 6]
```diff
@@ -9,13 +9,17 @@
 package salesforce.ui.pages.lightning.contracts;
 
 import java.util.HashMap;
+import java.util.Map;
+
```
Comment (SaulCM) [inline src/main/java/salesforce/ui/pages/lightning/contracts/NewContractPage.java:13]: Blank line.

File: src/main/java/salesforce/ui/pages/lightning/worktype/NewWorkTypePage.java [hunk 7]
```diff
@@ -40,6 +46,7 @@ public void setInputField(final String fieldName, final String fieldValue) {
             String xpath = String.format("//*[contains(text(),'%s')]/../..//*[@type='text']", fieldName);
             webElementAction.setInputField(By.xpath(xpath), fieldValue);
         }
+        webElementAction.clickByXpath("//div[@class='actionsContainer']");
```
Comment (SaulCM) [inline src/main/java/salesforce/ui/pages/lightning/worktype/NewWorkTypePage.java:49]: Could this locator be set as an attribute?

File: src/test/java/org/salesforce/steps/featureSteps.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+import org.testng.Assert;
+import salesforce.config.EnvConfig;
+import salesforce.ui.PageTransporter;
+import salesforce.ui.pages.LoginPage;
+import salesforce.ui.pages.lightning.HomePage;
+import salesforce.utlis.strategy.ICreatedFeature;
+import salesforce.utlis.strategy.IFeatureNew;
+import salesforce.utlis.strategy.IFeaturesPage;
+import salesforce.utlis.strategy.MapPages;
+
```
Comment (SaulCM) [inline src/test/java/org/salesforce/steps/featureSteps.java:16]: Remember to delete blank lines after formatting.

File: src/test/java/org/salesforce/hooks/generalHooks.java (hunk 3 above)
Comment (GustavoZHA) [reply src/test/java/org/salesforce/hooks/generalHooks.java:16]: I forgot thanks

File: src/main/java/salesforce/utlis/strategy/IFeaturesPage.java (hunk 4 above)
Comment (GustavoZHA) [reply src/main/java/salesforce/utlis/strategy/IFeaturesPage.java:8]: oksss

File: src/main/java/salesforce/ui/pages/lightning/worktype/NewWorkTypePage.java (hunk 7 above)
Comment (GustavoZHA) [reply src/main/java/salesforce/ui/pages/lightning/worktype/NewWorkTypePage.java:49]: I will refactor in the next pr

**Review by SaulCM — APPROVED** [review]

**Review by cristianfchq — APPROVED** [review]

**Review by JorgeCaceres-Jala — APPROVED** [review]

**Pull request merged**